x/marker/keeper/msg_server.go

```diff
@@ -270,6 +270,20 @@ func (k msgServer) Mint(goCtx context.Context, msg *types.MsgMintRequest) (*type
 		return nil, sdkerrors.ErrInvalidRequest.Wrap(err.Error())
 	}
 
+	if len(msg.Recipient) > 0 {
+		recipient, err := sdk.AccAddressFromBech32(msg.Recipient)
+		if err != nil {
+			return nil, sdkerrors.ErrInvalidAddress.Wrapf("invalid recipient: %s", msg.Recipient)
+		}
+		if recipient.Empty() {
+			return nil, sdkerrors.ErrInvalidAddress.Wrap("recipient address is zero or empty")
+		}
```

Comment:
This check can be removed. `AccAddressFromBech32` will either return a non-empty address, or an error.